contents/docs/llm-analytics/_snippets/trace-event.mdx

```diff
@@ -0,0 +1,47 @@
+A trace is a group that contains multiple spans, generations, and embeddings. Traces can be manually sent as events or appear as pseudo-events automatically created from child events.
+
+Event Name: `$ai_trace`  
+
+| Property | Description |
+|----------|-------------|
+| `$ai_trace_id` | The trace ID (a UUID to group related AI events together)<br/>Must contain only letters, numbers, and special characters: `-`, `_`, `~`, `.`, `@`, `(`, `)`, `!`, `'`, `:`, <code>\|</code><br/>Example: `d9222e05-8708-41b8-98ea-d4a21849e761` |
+| `$ai_input_state` | The input of the whole trace<br/>Example: `[{"role": "user", "content": "What's the weather in SF?"}]` or any JSON-serializable state |
+| `$ai_output_state` | The output of the whole trace<br/>Example: `[{"role": "assistant", "content": "The weather in San Francisco is..."}]` or any JSON-serializable state |
+| `$ai_latency` | **Optional**<br/>The latency of the trace in seconds |
+| `$ai_span_name` | **Optional**<br/>The name of the trace<br/>Example: `chat_completion`, `rag_pipeline` |
+| `$ai_is_error` | **Optional**<br/>Boolean to indicate if the trace encountered an error |
+| `$ai_error` | **Optional**<br/>The error message or object if the trace failed |
+
+### Example
+
+```bash
+curl -X POST "<ph_client_api_host>/i/v0/e/" \
+     -H "Content-Type: application/json" \
+     -d '{
+         "api_key": "<ph_project_api_key>",
+         "event": "$ai_trace",
+         "properties": {
+             "distinct_id": "user_123",
+             "$ai_trace_id": "d9222e05-8708-41b8-98ea-d4a21849e761",
+             "$ai_input_state": [{"role": "user", "content": "Tell me a fun fact about hedgehogs"}],
+             "$ai_output_state": [{"role": "assistant", "content": "Hedgehogs are small mammals with spines on their back."}],
+             "$ai_latency": 1.23,
+             "$ai_span_name": "hedgehog_facts_chat",
+             "$ai_is_error": false
+         },
+         "timestamp": "2025-01-30T12:00:00Z"
+     }'
+```
+
+### Pseudo-trace Events
```

Comment:
This seems like it should be above example. I think the example being at the end should be consistent. I only saw this section on the third time reading this. I stopped reading on example normally